spec/serializers/classification_serializer_spec.rb

```diff
@@ -33,21 +33,26 @@
     context "project context with last_id param present" do
       let(:project) { classification.project }
       let(:last_id) { classification.id }
+      let(:scope) do
+        Classification.where(project_id: project.id).after_id(last_id)
+      end
+      let(:prefix) { "/classifications?last_id=" }
+      let(:suffix) { "&page_size=1&project_id=#{project.id}" }
 
-      it "should insert the next hightest last_id into the next_href" do
+      it "should insert the highest page set id into the next_href" do
         second = create(:classification, project: project)
         params = {project_id: project.id, last_id: last_id, page_size: 1}
-        result = ClassificationSerializer.page(params, Classification.all, {})
+        result = ClassificationSerializer.page(params, scope, {})
```

Comment:
Using the `.where().after_id()` clause in your scope but also having the `last_id` in the params seems kind of redundant. The params are what the serializer uses and is what's being tested here.

Reply:
The serializer doesn't create the scope based on params, the scope is fed in via the controller action, so the project scope params here don't do anything, i'll remove them to clarify what's being tested.

Reply:
Ah - both the scope setup and the params are needed. Scope is what the params will create via normal controller flows and `scope_for` so this needs to be fed into the serializer. Also the params are used to construct valid next / prev href links and they flow normally through the serializer as they are here.